apache2/re_operators.c

```diff
@@ -2619,12 +2656,14 @@ static int msre_op_endsWith_execute(modsec_rec *msr, msre_rule *rule, msre_var *
 /* strmatch */
 
 static int msre_op_strmatch_param_init(msre_rule *rule, char **error_msg) {
+    assert(rule != NULL);
+    assert(rule->ruleset != NULL);
+    assert(error_msg != NULL);
     const apr_strmatch_pattern *compiled_pattern;
     char *processed = NULL;
     const char *pattern = rule->op_param;
     unsigned short int op_len;
 
-    if (error_msg == NULL) return -1;
```

Comment:
This check is removed, but I don't see the logic here: first the code checks if it NULL, returns if it is. Then sets it to NULL...

Reply:
we check the pointer and we assign the content of the pointer